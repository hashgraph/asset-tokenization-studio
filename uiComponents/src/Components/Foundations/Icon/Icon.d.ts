import type { IconProps as ChakraIconProps } from "@chakra-ui/icon";
import type { IconProps as PhosphorIconProps } from "@phosphor-icons/react";
import React from "react";
import type { CustomIconProps } from "./CustomIcon";
import type { ComponentWithAs } from "@chakra-ui/system";
export type IconButtonType = React.ForwardRefExoticComponent<PhosphorIconProps & React.RefAttributes<SVGSVGElement>>;
export interface BaseIconProps extends Omit<ChakraIconProps, "name" | "css" | "weight"> {
    size?: string;
    variant?: string;
}
export type ConditionalIconProps = {
    name?: false;
    as: any;
} | {
    name: CustomIconProps["name"];
    as?: never;
};
export type IconProps = BaseIconProps & ConditionalIconProps;
export declare const Icon: ComponentWithAs<"svg", BaseIconProps & IconProps>;
