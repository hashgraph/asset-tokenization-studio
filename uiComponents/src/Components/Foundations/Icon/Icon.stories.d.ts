import React from "react";
declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.SVGProps<SVGSVGElement>, any, import("./Icon").BaseIconProps & import("./Icon").IconProps, import("@chakra-ui/system/dist/system.types").As>>;
export default meta;
export declare const RemixIcon: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.SVGProps<SVGSVGElement>, any, import("./Icon").BaseIconProps & import("./Icon").IconProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const CustomIconStory: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.SVGProps<SVGSVGElement>, any, import("./Icon").BaseIconProps & import("./Icon").IconProps, import("@chakra-ui/system/dist/system.types").As>>;
