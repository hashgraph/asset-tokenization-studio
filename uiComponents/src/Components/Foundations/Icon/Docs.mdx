import {
  Story,
  Meta,
  Title,
  Controls,
  Primary,
  Stories,
  ArgTypes,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as IconStories from "./Icon.stories";

<Meta of={IconStories} />

<Title />
<Primary />

## Props

<ArgTypes of={IconStories} />

## Stories

<Stories title="" />
