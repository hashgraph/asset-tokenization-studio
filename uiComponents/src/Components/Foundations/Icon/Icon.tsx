import type { IconProps as ChakraIconProps } from "@chakra-ui/icon";
import type { IconProps as PhosphorIconProps } from "@phosphor-icons/react";
import { Icon as ChakraIcon } from "@chakra-ui/icon";
import { forwardRef } from "@chakra-ui/system";
import React from "react";
import type { CustomIconProps } from "./CustomIcon";
import { CustomIcon } from "./CustomIcon";
import type { ComponentWithAs } from "@chakra-ui/system";

export type IconButtonType = React.ForwardRefExoticComponent<
  PhosphorIconProps & React.RefAttributes<SVGSVGElement>
>;

export interface BaseIconProps
  extends Omit<ChakraIconProps, "name" | "css" | "weight"> {
  size?: string;
  variant?: string;
}

export type ConditionalIconProps =
  | { name?: false; as: any }
  | { name: CustomIconProps["name"]; as?: never };

export type IconProps = BaseIconProps & ConditionalIconProps;

export const Icon: ComponentWithAs<"svg", BaseIconProps & IconProps> =
  forwardRef<BaseIconProps & IconProps, "svg">(
    ({ name, ...props }: BaseIconProps & IconProps, ref) => {
      if (name) {
        return <CustomIcon ref={ref} name={name} {...props} />;
      }
      return <ChakraIcon ref={ref} {...props} />;
    }
  );
