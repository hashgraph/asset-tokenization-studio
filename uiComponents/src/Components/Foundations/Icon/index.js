export * from "./Icon";
export * from "./CustomIcon";
