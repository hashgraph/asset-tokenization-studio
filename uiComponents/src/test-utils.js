import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
