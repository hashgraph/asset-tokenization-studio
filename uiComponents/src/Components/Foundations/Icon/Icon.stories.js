var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { icons } from "@/Theme/icons";
import React from "react";
import { Icon } from "./Icon";
import { UserRectangle } from "@phosphor-icons/react";
var CustomNameIcons = Object.keys(icons);
var meta = {
    title: "Design System/Foundations/Icon",
    component: Icon,
    argTypes: {
        name: {
            options: CustomNameIcons
        },
        size: {
            control: "select",
            options: ["sm", "md"]
        }
    },
    parameters: {
        design: {
            type: "figma",
            url: "https://www.figma.com/file/yn0TWBlB8x5WbsEY8BdML7/Palm-Design-System?node-id=1353%3A7966"
        },
        docs: {}
    }
};
export default meta;
var Template = function (args) {
    return React.createElement(Icon, __assign({}, args));
};
export var RemixIcon = Template.bind({});
RemixIcon.args = {
    as: UserRectangle
};
export var CustomIconStory = Template.bind({});
CustomIconStory.args = {
    name: "Progress"
};
CustomIconStory.argTypes = {
    name: {
        control: {
            type: "select",
            options: CustomNameIcons
        }
    }
};
