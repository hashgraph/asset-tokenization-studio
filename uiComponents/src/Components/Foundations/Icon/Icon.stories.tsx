import { icons } from "@/Theme/icons";

import type { Meta, StoryFn } from "@storybook/react";
import React from "react";
import { Icon } from "./Icon";
import { UserRectangle } from "@phosphor-icons/react";

const CustomNameIcons = Object.keys(icons);
const meta = {
  title: "Design System/Foundations/Icon",
  component: Icon,
  argTypes: {
    name: {
      options: CustomNameIcons,
    },
    size: {
      control: "select",
      options: ["sm", "md"],
    },
  },
  parameters: {
    design: {
      type: "figma",
      url: "https://www.figma.com/file/yn0TWBlB8x5WbsEY8BdML7/Palm-Design-System?node-id=1353%3A7966",
    },
    docs: {},
  },
} as Meta<typeof Icon>;
export default meta;

const Template: StoryFn<typeof Icon> = (args) => {
  return <Icon {...args} />;
};

export const RemixIcon = Template.bind({});
RemixIcon.args = {
  as: UserRectangle,
};

export const CustomIconStory = Template.bind({});
CustomIconStory.args = {
  name: "Progress",
};
CustomIconStory.argTypes = {
  name: {
    control: {
      type: "select",
      options: CustomNameIcons,
    },
  },
};
