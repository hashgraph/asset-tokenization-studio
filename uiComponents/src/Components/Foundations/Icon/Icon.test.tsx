import { render } from "@/test-utils";
import React from "react";
import type { CustomIconProps } from "./CustomIcon";
import { Icon } from "./Icon";
import { Moon } from "@phosphor-icons/react";

describe(`<Icon />`, () => {
  const customIcon = "Progress";

  const factoryComponent = (name?: CustomIconProps["name"], as?: any) =>
    render(<Icon name={name} as={as} />);

  test("renders correctly when uses custom icon", () => {
    const component = factoryComponent(customIcon);

    expect(component.asFragment()).toMatchSnapshot("Using custom icon");
  });

  test("renders correctly when uses remix icon", () => {
    const component = factoryComponent(undefined, Moon);

    expect(component.asFragment()).toMatchSnapshot("Using remix icon");
  });

  test("renders correctly when no icon exists", () => {
    expect(() => factoryComponent("other")).toThrowError();
  });

  test("renders correctly when an icon remix is replaced by a custom icon remix", () => {
    const themeCustom = {
      icons: {
        BankFill: {
          viewBox: "0 0 22 22",
          path: [
            <path
              d="M18.5 11C18.5 13.24 17.5186 15.2501 15.9608 16.6251C14.6384 17.7923 12.9024 18.5 11 18.5C6.85786 18.5 3.5 15.1421 3.5 11C3.5 9.05175 4.2422 7.27801 5.46013 5.94425C6.83231 4.44158 8.80584 3.5 11 3.5C15.1421 3.5 18.5 6.85786 18.5 11Z"
              stroke="#0F172A"
            />,
            <path
              d="M10.1668 12.3216L13.9968 8.49121L14.5864 9.08038L10.1668 13.5L7.51514 10.8483L8.1043 10.2591L10.1668 12.3216Z"
              fill="#0F172A"
            />,
            <circle cx="11" cy="11" r="10.5" stroke="#0F172A" />,
          ],
          defaultProps: {
            fill: "none",
          },
        },
      },
    };

    const component = render(<Icon name={"BankFill"} />, themeCustom);

    expect(component.asFragment()).toMatchSnapshot(
      "Using replacing customIcon"
    );
  });
});
