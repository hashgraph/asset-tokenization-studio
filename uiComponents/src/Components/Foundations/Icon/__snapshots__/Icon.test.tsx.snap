// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders correctly when an icon remix is replaced by a custom icon remix: Using replacing customIcon 1`] = `
<DocumentFragment>
  <svg
    class="chakra-icon css-nyt2fo"
    focusable="false"
    viewBox="0 0 22 22"
  >
    <path
      d="M18.5 11C18.5 13.24 17.5186 15.2501 15.9608 16.6251C14.6384 17.7923 12.9024 18.5 11 18.5C6.85786 18.5 3.5 15.1421 3.5 11C3.5 9.05175 4.2422 7.27801 5.46013 5.94425C6.83231 4.44158 8.80584 3.5 11 3.5C15.1421 3.5 18.5 6.85786 18.5 11Z"
      stroke="#0F172A"
    />
    <path
      d="M10.1668 12.3216L13.9968 8.49121L14.5864 9.08038L10.1668 13.5L7.51514 10.8483L8.1043 10.2591L10.1668 12.3216Z"
      fill="#0F172A"
    />
    <circle
      cx="11"
      cy="11"
      r="10.5"
      stroke="#0F172A"
    />
  </svg>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Icon /> renders correctly when uses custom icon: Using custom icon 1`] = `
<DocumentFragment>
  <svg
    class="chakra-icon css-onkibi"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      clip-rule="evenodd"
      d="M2.04762 12C2.04762 6.50345 6.50345 2.04762 12 2.04762C17.4965 2.04762 21.9524 6.50345 21.9524 12C21.9524 17.4965 17.4965 21.9524 12 21.9524C6.50345 21.9524 2.04762 17.4965 2.04762 12ZM12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1ZM6.71213 7.17424C8.02194 5.73987 9.90561 4.8412 11.9999 4.8412C15.9536 4.8412 19.1587 8.04627 19.1587 11.9999C19.1587 14.138 18.222 16.0566 16.735 17.3691C15.4727 18.4832 13.8158 19.1587 11.9999 19.1587C8.04627 19.1587 4.8412 15.9536 4.8412 11.9999C4.8412 10.1403 5.54958 8.44734 6.71213 7.17424ZM11.9999 3.79358C9.59909 3.79358 7.43846 4.82523 5.93852 6.46781C4.60661 7.92638 3.79358 9.86881 3.79358 11.9999C3.79358 16.5322 7.46769 20.2063 11.9999 20.2063C14.0808 20.2063 15.9819 19.4311 17.4282 18.1546C19.131 16.6516 20.2063 14.451 20.2063 11.9999C20.2063 7.46769 16.5322 3.79358 11.9999 3.79358Z"
      fill="#0F172A"
      fill-rule="evenodd"
      opacity="0.2"
    />
    <path
      clip-rule="evenodd"
      d="M2.04762 12C2.04762 6.50345 6.50345 2.04762 12 2.04762C17.4965 2.04762 21.9524 6.50345 21.9524 12C21.9524 17.4965 17.4965 21.9524 12 21.9524C11.7107 21.9524 11.4762 22.1869 11.4762 22.4762C11.4762 22.7655 11.7107 23 12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 12.2893 1.23452 12.5238 1.52381 12.5238C1.8131 12.5238 2.04762 12.2893 2.04762 12ZM16.7353 17.3692C15.473 18.4833 13.8161 19.1587 12.0002 19.1587C8.04658 19.1587 4.84151 15.9537 4.84151 12C4.84151 9.97288 5.68338 8.14328 7.03798 6.84021C7.24647 6.63965 7.2529 6.30805 7.05234 6.09957C6.85178 5.89108 6.52019 5.88465 6.3117 6.08521C4.7603 7.5776 3.79389 9.67639 3.79389 12C3.79389 16.5323 7.46799 20.2064 12.0002 20.2064C14.0811 20.2064 15.9822 19.4312 17.4285 18.1546C17.6454 17.9632 17.6661 17.6322 17.4746 17.4153C17.2832 17.1984 16.9522 17.1778 16.7353 17.3692Z"
      fill="#0F172A"
      fill-rule="evenodd"
    />
  </svg>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Icon /> renders correctly when uses remix icon: Using remix icon 1`] = `
<DocumentFragment>
  <svg
    class="chakra-icon css-13otjrl"
    fill="currentColor"
    focusable="false"
    height="1em"
    viewBox="0 0 256 256"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M233.54,142.23a8,8,0,0,0-8-2,88.08,88.08,0,0,1-109.8-109.8,8,8,0,0,0-10-10,104.84,104.84,0,0,0-52.91,37A104,104,0,0,0,136,224a103.09,103.09,0,0,0,62.52-20.88,104.84,104.84,0,0,0,37-52.91A8,8,0,0,0,233.54,142.23ZM188.9,190.34A88,88,0,0,1,65.66,67.11a89,89,0,0,1,31.4-26A106,106,0,0,0,96,56,104.11,104.11,0,0,0,200,160a106,106,0,0,0,14.92-1.06A89,89,0,0,1,188.9,190.34Z"
    />
  </svg>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
